torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -55,6 +55,24 @@ struct TORCH_API AccumulateGrad : public Node {
         // holding references to the Tensor and that is fine since these are not
         // exposed to the user.
         update_grad(new_grad_copy.detach());
+      } else if (
+          !GradMode::is_enabled() && new_grad_copy.is_sparse() &&
+          new_grad_copy._indices().is_contiguous() &&
+          new_grad_copy._values().is_contiguous() &&
```

Comment:
@albanD Maybe I missed something obvious. We do shallow copy only when indices and values are contiguous?

Reply:
Yes. If they are already contiguous and not used anywhere else, it is fine to reuse the same memory.
Otherwise, we use clone to get brand new memory.